conans/client/command.py

```diff
@@ -1222,7 +1228,24 @@ def upload(self, *args):
 
         args = parser.parse_args(*args)
 
-        if args.query and args.package:
+        try:
+            pref = PackageReference.loads(args.pattern_or_reference, validate=True)
+            reference = pref.ref.full_repr()
+            package_id = pref.id
+        except ConanException:
+            reference = args.pattern_or_reference
+            package_id = args.package
+
+            if package_id:
```

Comment:
I think it's not aligned with line 1243

Reply:
The `else` correspond to `try`, `except`.